vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,27 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+// HealthcheckResult stores information about a single run of a healthcheck probe
+type HealthcheckResult struct {
+	Start, End time.Time
+	ExitCode   int    // 0=healthy, 1=unhealthy, 2=starting, -1=error running probe
+	Output     string // Output from last check
+}
+
+// Health states
+const (
+	Starting  = "starting"  // Container is not yet ready
```

Comment:
If a container transitions to `healthy`, then fails one health check, what is the next state to report? If a health check is run several times, how does it know its last state?